server/src/main/kotlin/suwayomi/tachidesk/manga/impl/sync/KoreaderSyncService.kt

```diff
@@ -0,0 +1,359 @@
+                                Map::class.java,
+                            )["message"]
+                        }.getOrNull()
+                    val finalMessage = errorMessage ?: "Registration failed with code ${response.code}"
+                    AuthResult(false, finalMessage)
+                }
+            }
+        } catch (e: Exception) {
+            logger.error(e) { "[KOSYNC REGISTER] Exception" }
+            AuthResult(false, e.message)
+        }
+    }
+
+    private suspend fun authorize(
+        username: String,
+        userkey: String,
+    ): AuthResult {
+        val request =
+            buildRequest("${serverConfig.koreaderSyncServerUrl.value.removeSuffix("/")}/users/auth") {
+                get()
+                addHeader("x-auth-user", username)
+                addHeader("x-auth-key", userkey)
+            }
+
+        return try {
+            network.client.newCall(request).execute().use { response ->
+                if (response.isSuccessful) {
+                    AuthResult(true)
+                } else {
+                    val isUserNotFound = response.code == 401 // Unauthorized often means user/pass combo is wrong
+                    AuthResult(false, "Authorization failed with code ${response.code}", isUserNotFoundError = isUserNotFound)
+                }
+            }
+        } catch (e: Exception) {
+            logger.error(e) { "[KOSYNC AUTHORIZE] Exception" }
+            AuthResult(false, e.message)
+        }
+    }
+
+    suspend fun connect(
+        username: String,
+        password: String,
+    ): KoSyncConnectPayload {
+        val userkey = Hash.md5(password)
+        val authResult = authorize(username, userkey)
+
+        if (authResult.success) {
+            GlobalConfigManager.updateValue("server.koreaderSyncUsername", username)
+            GlobalConfigManager.updateValue("server.koreaderSyncUserkey", userkey)
+            return KoSyncConnectPayload(true, "Login successful.", username)
+        }
+
+        if (authResult.isUserNotFoundError) {
+            logger.info { "[KOSYNC CONNECT] Authorization failed, attempting to register new user." }
+            val registerResult = register(username, userkey)
+            return if (registerResult.success) {
+                GlobalConfigManager.updateValue("server.koreaderSyncUsername", username)
+                GlobalConfigManager.updateValue("server.koreaderSyncUserkey", userkey)
+                KoSyncConnectPayload(true, "Registration successful.", username)
+            } else {
+                KoSyncConnectPayload(false, registerResult.message ?: "Registration failed.", null)
+            }
+        }
+
+        return KoSyncConnectPayload(false, authResult.message ?: "Authentication failed.", null)
+    }
+
+    suspend fun logout() {
+        GlobalConfigManager.updateValue("server.koreaderSyncUsername", "")
+        GlobalConfigManager.updateValue("server.koreaderSyncUserkey", "")
```

Comment:
This needs to set the values in serverConfig, not GlobalConfigManager
```suggestion
        serverConfig.koreaderSyncUsername.value = ""
        serverConfig.koreaderSyncUserkey.value = ""
```